src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -124,61 +124,70 @@ public synchronized void close() throws IOException {
         }
     }
 
+    @SneakyThrows
     @SuppressFBWarnings("SWL_SLEEP_WITH_LOCK_HELD")
     @SuppressWarnings("PMD.AvoidDeeplyNestedIfStmts")
     private void stopGracefully() {
         int pid = ((ProcessImplForWin32) process).getPid();
         boolean sentConsoleCtrlEvent = false;
         synchronized (Kernel32.INSTANCE) {
-            Kernel32 k32 = Kernel32.INSTANCE;
-            // if the console is already attached, a second attach call will fail.
-            if (!k32.FreeConsole()) {
-                logger.error("FreeConsole error {}", k32.GetLastError());
-            }
-            if (!k32.AttachConsole(pid)) {
-                logger.error("AttachConsole error {}", k32.GetLastError());
-                // Console already attached so we cannot signal it
-                if (k32.GetLastError() == ERROR_ACCESS_DENIED) {
```

Comment:
We just did FreeConsole. This should be redundant. This will hit `return` and go to `finally` which does the cleanup.